### PR TITLE
Armageddon 2

### PR DESCRIPTION
**Pull request by Przemog1**

### CODE REVIEW

File: src/irr/ext/MitsubaLoader/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	${IRR_EXT_INTERNAL_INCLUDE_DIR}/CElementEmitter.h
+	${IRR_EXT_INTERNAL_INCLUDE_DIR}/CElementFactory.h
+	${IRR_EXT_INTERNAL_INCLUDE_DIR}/CGlobalMitsubaMetadata.h
+	${IRR_EXT_INTERNAL_INCLUDE_DIR}/ParserUtil.h
+	${IRR_EXT_INTERNAL_INCLUDE_DIR}/CSerializedLoader.h
+	${IRR_EXT_INTERNAL_INCLUDE_DIR}/CMitsubaLoader.h
+	${IRR_EXT_INTERNAL_INCLUDE_DIR}/CMitsubaMaterialCompilerFrontend.h
+)
+
+set(IRR_EXT_MITSUBA_LOADER_SRC
```
Comment (devshgraphicsprogramming) [inline src/irr/ext/MitsubaLoader/CMakeLists.txt]: IRR prefix needs fixing

File: src/irr/ext/OptiX/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	"${IRR_ROOT_PATH}/3rdparty"
+	"${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}"
+	"${OPTIX_INCLUDE_DIR}"
+)
+
+irr_create_ext_library_project(
+	OPTIX
+	"${IRR_EXT_OPTIX_H}"
+	"${IRR_EXT_OPTIX_SRC}"
+	"${IRR_EXT_OPTIX_EXTERNAL_INCLUDE}"
```
Comment (devshgraphicsprogramming) [inline src/irr/ext/OptiX/CMakeLists.txt:27]: irr prefix

File: src/irr/ext/RadeonRays/CMakeLists.txt [hunk 3]
```diff
@@ -1,8 +1,8 @@
-include(../../cmake/common.cmake)
-
+include(${IRR_ROOT_PATH}/cmake/common.cmake)
+set(IRR_EXT_INTERNAL_INCLUDE_DIR "${IRR_ROOT_PATH}/include/irr/ext/RadeonRays")
```
Comment (devshgraphicsprogramming) [inline src/irr/ext/RadeonRays/CMakeLists.txt:2]: IRR prefix

File: include/irr/ext/Readme.md [hunk 4]
```diff
@@ -4,11 +4,11 @@ Stuff which is not 100% necessary for a GPGPU engine, or could have 3 or 4 possi
...
+4) Keep all your header files in `./include/irr/ext/`, source files in `./src/irr/ext/`
+5) Include your own files (even in your own .c/cpp files) by specifying their paths relative to the irrlicht root (see rule 2). So instead of your extension's source files (and the users source files) asking for `#include "YourHeader.h"` they should ask for `#include "../../include/irr/ext/YourExtension/YourHeader.h"
 6) Provide a CMakeLists.txt
 7) Add a `IRR_BUILD_YOUR_EXTENION` CMake option that controls the building of any examples using Your Extension and that it is set to OFF by default. This is to make sure that your extension does not harm everybody's build-time and should Your Extension's dependencies fail to pull/configure it doesn't affect the rest of the library.
 8) Make sure your extension's PUBLIC headers (if its a header only or a source inclusion only extension then all headers) are installed to `${CMAKE_INSTALL_PREFIX}/include/irr/ext/YourExtension/`
```
Comment (devshgraphicsprogramming) [inline include/irr/ext/Readme.md:14]: change irr to nbl and irrlicht to nabla in the readme

File: src/irr/ext/Bullet/CMakeLists.txt [hunk 5]
```diff
@@ -7,17 +7,17 @@ file(GLOB_RECURSE BULLETCOLLISION_H ${BULLET_INCLUDE_PATH}/BulletCollision/*.h)
...
-  BulletUtility.h 
-  CDebugRender.h 
-  CInstancedMotionState.h
-  CPhysicsWorld.h
-  IMotionStateBase.h
+  ${IRR_EXT_INTERNAL_INCLUDE_DIR}/BulletUtility.h 
+  ${IRR_EXT_INTERNAL_INCLUDE_DIR}/CDebugRender.h 
+  ${IRR_EXT_INTERNAL_INCLUDE_DIR}/CInstancedMotionState.h
+  ${IRR_EXT_INTERNAL_INCLUDE_DIR}/CPhysicsWorld.h
+  ${IRR_EXT_INTERNAL_INCLUDE_DIR}/IMotionStateBase.h
```
Comment (devshgraphicsprogramming) [inline src/irr/ext/Bullet/CMakeLists.txt:20]: irr prefix

File: include/EDeviceTypes.h [hunk 6]
```diff
@@ -28,7 +28,7 @@ namespace irr
 
 		//! A device which uses Simple DirectMedia Layer
 		/** The SDL device works under all platforms supported by SDL but first must be compiled
-		in by defining the IRR_USE_SDL_DEVICE macro in IrrCompileConfig.h */
+		in by defining the NBL_USE_SDL_DEVICE macro in IrrCompileConfig.h */
 		EIDT_SDL,
```
Comment (devshgraphicsprogramming) [inline include/EDeviceTypes.h:32]: you can remove this enum actually.

File: include/irr/ext/MitsubaLoader/CElementIntegrator.h [hunk 7]
```diff
@@ -37,7 +37,7 @@ class CElementIntegrator : public IElement
 			ADJ_P_TRACER,
 			ADAPTIVE,
 			VPL,
-			IRR_CACHE,
```
Comment (devshgraphicsprogramming) [inline include/irr/ext/MitsubaLoader/CElementIntegrator.h:40]: actually it means "irradiance cache" so no need to change ;)

File: README.md [hunk 8]
```diff
@@ -1,8 +1,8 @@
...
+**Nabla** is a new renovated version of older **[Irrlicht](http://irrlicht.sourceforge.net/)** engine. The project currently aims for a thread-able and *Vulkan*-centered API, but currently works on *OpenGL* only. This framework has been kindly begun by the founder ***[@devshgraphicsprogramming](https://github.com/devshgraphicsprogramming)*** of **[Devsh Graphics Programming Sp. z O.O.](http://devsh.eu/)**  and almost entirely sponsored by **Build A World Aps**. in it's early days, and now picked up by the **[Ditt](https://www.ditt.nl/)** company. The `stable-ish` branch is used in production releases of **[Build A World EDU](https://edu.buildaworld.net/)**, since 2015. The framework has been used both for game development and ArchViz.
```
Comment (devshgraphicsprogramming) [inline README.md:5]: This needs mentioning the old IrrlichtBaW repo

File: README.md [hunk 9]
```diff
@@ -1,8 +1,8 @@
...
 
 ## Contracting
```
Comment (devshgraphicsprogramming) [inline README.md:8]: change the list of specializations to bulletpoints

Comment (devshgraphicsprogramming) [reply README.md:8]: Also all the screenshots/gifs come from `![](https://github.com/buildaworldnet/IrrlichtBAW/blob/` need URL update

File: README.md [hunk 10]
```diff
@@ -18,7 +18,7 @@ Contact ***[@devshgraphicsprogramming](https://github.com/devshgraphicsprogrammi
 
 
 
-![](https://github.com/buildaworldnet/IrrlichtBAW/blob/ad02c69e384c6655951c99d3b4bee5178a9dab2f/site_media/readme/gifs/myballs/Multiple%20Importance%20Sampling%20and%20Depth%20of%20Field%202.gif?raw=true)
+![](https://github.com/Devsh-Graphics-Programming/Nabla/blob/ad02c69e384c6655951c99d3b4bee5178a9dab2f/site_media/readme/gifs/myballs/Multiple%20Importance%20Sampling%20and%20Depth%20of%20Field%202.gif?raw=true)
```
Comment (devshgraphicsprogramming) [inline README.md:21]: there's more images.

Comment (Przemog1) [reply README.md:21]: I know xd

File: README.md (hunk 8 above)
Comment (devshgraphicsprogramming) [reply README.md:5]: Nabla (previously called [IrrlichtBaW](http to old IRrlichtBaw repo) ....


at the end write that "The name change to Nabla allows for using Nabla side by side with the legacy Irrlicht and IrrlichtBaW engines.*

**Pull request merged**